### PR TITLE
Almbic exporter now exports visibility

### PR DESCRIPTION
**Pull request by LegacyID1991**
This is related to #55 

changes has been made to:
> [_extract_animation.py_](https://github.com/mindbender-studio/core/blob/master/mindbender/maya/plugins/extract_animation.py#L50)
> [_lib.py_](https://github.com/mindbender-studio/core/blob/master/mindbender/maya/lib.py#L199)

Changes will result in that when **publishing** an animation from _PYBLISH_ the exporter now also writes the animated visiblilty to the exported alembic.

### CODE REVIEW

**Comment by mottosso** [comment]
Hi Arvid, I've put together a little tutorial on how to avoid getting all of those commits in the commit history every time you make a PR.

[arvidguide.zip](https://github.com/mindbender-studio/core/files/801872/arvidguide.zip)

It'll enable you to work on features individually and have better control over what is in your PR. Let me know if anything is unclear.

**Comment by LegacyID1991** [comment]
I watched it, no real uncertainties now but there will probably be some the first couple of times.
Do you want me to take this PR down and make a new one ❓ 
Or can we merge this one and I will make the next PR much cleaner ❔

File: mindbender/maya/lib.py [hunk 1]
```diff
@@ -166,6 +166,7 @@ def export_alembic(nodes,
                    file,
                    frame_range=None,
                    uv_write=True,
+                   write_Visibility=True,
```
Comment (mottosso) [inline mindbender/maya/lib.py]: Use all lowercase please, so as to conform to the surrounding naming convention - a.k.a. snake_case.

Comment (mottosso) [inline mindbender/maya/lib.py]: I'm also thinking we avoid the bad naming convention used by the original `AbcExport` and prefix each of the 6 optional outputs with `write_`, including `uv_write`.

The 6 optionals being:

- uvWrite
- writeColorSets
- writeFaceSets
- writeVisibility
- writeCreases

Let's rename `uv_write` to `write_uv` and make `write_Visibility` into `write_visibility`. What do you think?

Comment (LegacyID1991) [inline mindbender/maya/lib.py]: you are right, it will look much better after these changes, I am on it

**Comment by LegacyID1991** [comment]
how is that?

**Comment by mottosso** [comment]
Looks good to me. Have a look at getting a test in there. I can help you tomorrow if you need.

**Comment by mottosso** [comment]
Perfect.

Ok, so in order to make the test, I couldn't rely on hardcoding of a path like in the [original example](https://github.com/mindbender-studio/core/issues/55#issuecomment-282494913). Although that was a very good way of getting a first draft out so do keep doing that if you need to.

Instead, in order for the test to last, we need to use the abstractions provided by the pipeline in order to (1) hunt down the published asset and (2) import it into Maya.

In order not to depend on hardcoding the path, I needed the loader to give me something back. A result of the importing, so that I could use that to query whether or not the export was successful. This was where #58 came up. Before, loading an asset returned nothing, which made it difficult to accurately find the loaded Alembic and query its visibility.

So, if all that makes sense, I'm happy with this and can merge if you are too.

**Comment by LegacyID1991** [comment]
So now then this PR resolves both #58 and #55 
and now with [this](https://github.com/mindbender-studio/core/pull/57/commits/6f4a2cc3b3d0417c2cd05c13b0b1991cca2f29e7) we have a working test ?

**Comment by mottosso** [comment]
Exactly!

**Comment by LegacyID1991** [comment]
**UPDATE**
This PR is related to #55 and #58 

changes has been made to:

> [_extract_animation.py_](https://github.com/mindbender-studio/core/blob/master/mindbender/maya/plugins/extract_animation.py#L50)
> [_lib.py_](https://github.com/mindbender-studio/core/blob/master/mindbender/maya/lib.py#L199)
> [_pipeline.py_](https://github.com/LegacyID1991/core/blob/a2065258f2e1efd295ffa1f31c3d639a5f9c061f/mindbender/maya/pipeline.py#L250)
> [_test_workflow.py_](https://github.com/LegacyID1991/core/blob/master/mindbender/maya/tests/test_workflow.py)

Changes will result in that when publishing an animation from PYBLISH the exporter now also writes the animated visiblilty to the exported alembic.
Implements a test to be run by Travis CI to test the integrity of this new addition and future changes to this plugin

**Pull request merged**